### DISCUSSION TITLE
RESEARCH: nixBlitz

### DISCUSSION THREAD

**Discussion by rootzoll** [raspiblitz/raspiblitz#3903] [Research]
More and more people ask about using nixOS as a base for RaspiBlitz - but that would be a complete different approach on how to install & config the system. But maybe we reached a point in the development of RaspiBlitz where this can spawn a sister project: nixBlitz - with v1.8.0 we introdiced the BackedAPI by @fusion44 and the WebUI by @cstenglein - those offer an essential but minimal subset of the features the SSH RaspiBlitz is using. The backendAPI has the goal not to be fixed on the RaspiBlitz system and could cater to any generic bitcoin/ln/cln based system - delivering services to the WebUI. Thats seems like the perfect abstraction point to kind of swap engines underneath the backendAPI with a nixOS based system. Having both in parallel RaspiBlitz & nixBlitz - not with the final goal but the possibility for users to migrate at one point in the future. 
  
![Bildschirmfoto 2022-09-26 um 13 41 39](https://user-images.githubusercontent.com/2472225/192280428-0a77a190-3ed0-4d46-a102-b389cf4716c4.png)

This issue is a collector of reasearch ind ideas in that direction. Subscripbe on this issue I you like to join the conversation.

Similar idea was also mentioned here:
https://github.com/rootzoll/raspiblitz/issues/1681#issuecomment-852154427

**Comment by chrisguida** [comment]
Is there a chat group or somewhere where I can join the effort here? Let's make this happen!

**Comment by chrisguida** [comment]
Seems like the work was being done here: https://github.com/MatthewCroughan/nixblitz

How can I test?

**Comment by chrisguida** [comment]
Actually seems [MYBONK](https://mybonk.github.io/mybonk-wiki/) is a successor to this initiative (via https://twitter.com/rootzoll/status/1694038550755352662)
